web/src/app/modules/views/main/editors/modules/contents-container/base/contents-container-base.ts

```diff
@@ -0,0 +1,58 @@
+import { SpecmateDataService } from '../../../../../../data/modules/data-service/services/specmate-data.service';
+import { NavigatorService } from '../../../../../../navigation/modules/navigator/services/navigator.service';
+import { TranslateService } from '@ngx-translate/core';
+import { IContainer } from '../../../../../../../model/IContainer';
+import { ConfirmationModal } from '../../../../../../notification/modules/modals/services/confirmation-modal.service';
+import { Id } from '../../../../../../../util/id';
+import { OnInit } from '@angular/core';
+
+export abstract class ContentContainerBase<T extends IContainer> implements OnInit {
+
+    protected abstract get newName(): string;
+    protected abstract get parent(): IContainer;
+    protected abstract set parent(parent: IContainer);
```

Comment:
In all implementing classes set parent does the same, maybe it should be pulled up to the base class instead of being abstract

Reply:
That does not work, because the set method must, in the end be a angular input. But in this base class, which is not a component, we cannot use the annotation.